src/components/MobileControls/MobileControls.tsx

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable react-hooks/rules-of-hooks */
```

Comment:
why is this necessary ?

Reply:
This is not necessary. Now I have removed this, and refactored the logic of the component